include/async/schedulers/runloop_scheduler.hpp

```diff
@@ -27,6 +28,57 @@
 #endif
 
 namespace async {
+#if HAS_CONDITION_VARIABLE
+template <typename Uniq, std::size_t N> struct synchronizer {
```

Comment:
question: should this be in a `detail` namespace?

Reply:
OK - I took it out of `_runloop`, but it makes sense for it to be in `detail`.